### PR TITLE
Using Search Fields from CLI & API

### PR DESCRIPTION
**Pull request by evanxg852000**
### Context / purpose
Closes https://github.com/quickwit-inc/quickwit/issues/225

### Description
Added `search fields` to rest API and making use of it in the CLI

### Checklist
*Remove or complete this list if required.*
- [x] tested locally
- [x] updated unit tests

### CODE REVIEW

**Review by fulmicoton — COMMENTED** [review]
See comment. 
We want to accept
```searchField=title&searchField=body```

File: quickwit-serve/src/rest.rs [hunk 1]
```diff
@@ -190,7 +194,7 @@ mod tests {
     async fn test_rest_search_api_route_simple_default_num_hits_default_offset() {
         let rest_search_api_filter = search_filter();
         let (index, req) = warp::test::request()
-            .path("/api/v1/quickwit-demo-index/search?query=*&endTimestamp=1450720000")
+            .path("/api/v1/quickwit-demo-index/search?query=*&endTimestamp=1450720000&searchFields[0]=title&searchFields[1]=body")
```
[the remark this answers is not in the document]
Comment (fulmicoton) [reply quickwit-serve/src/rest.rs:197]: Also the plural makes sense in the code but it is not user friendly.

Several occurrences of `searchField` might be better.

**Comment by evanxg852000** [comment]
So I tried the custom deserialization Impl but serde_qs keep complaining about duplicate keys even with the `flatten` attribute
For now, I am using a comma-separated list while searching a way to aggregate the values with same keys.

**Comment by fmassot** [comment]
I'm ok to support only comma separated value and handle the other syntax later.

**Review by fmassot — APPROVED** [review]

**Pull request merged**